app/views/admin/products_v3/_variant_row.html.haml

```diff
@@ -37,20 +37,24 @@
       = f.label :on_demand do
         = f.check_box :on_demand, 'data-action': 'change->toggle-control#disableIfPresent change->popout#closeIfChecked'
         = t(:on_demand)
-%td.col-producer.naked_inputs
+%td.col-producer.field.naked_inputs
   = render(SearchableDropdownComponent.new(form: f,
       name: :supplier_id,
       aria_label: t('.producer_field_name'),
       options: producer_options,
       selected_option: variant.supplier_id,
-      placeholder_value: t('admin.products_v3.filters.search_for_producers')))
+      include_blank: t('admin.products_v3.filters.select_producer'),
+      placeholder_value: t('admin.products_v3.filters.select_producer')))
+  = error_message_on variant, :supplier
```

Comment:
Great, thanks for adding `error_message_on`. It probably should have been added in the beginning.